src/features/AnnouncementModal/AnnouncementModal.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Flex,
+  Image,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  ModalBody,
+  Stack,
+  Link,
+} from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { ProgressIndicator } from "components/ProgressIndicator"
+import { useCallback, useMemo, useState } from "react"
+import { BiRightArrowAlt } from "react-icons/bi"
+
+import { useAnnouncements } from "hooks/useAnnouncement"
+
+import { Announcement } from "types/announcements"
+
+import { NewFeatureTag } from "./components/NewFeatureTag"
+
+interface AnnouncementModalProps {
+  isOpen: boolean
+  onClose: () => void
+  announcements: Announcement[]
+  link: string
+}
+
+export const AnnouncementModal = ({
+  isOpen,
+  announcements,
+  onClose,
+  link,
+}: AnnouncementModalProps): JSX.Element => {
+  const [currActiveIdx, setCurrActiveIdx] = useState<number>(0)
+  const { setLastSeenAnnouncement } = useAnnouncements()
+  const isLastAnnouncement = useMemo(
+    () => currActiveIdx === announcements.length - 1,
+    [announcements.length, currActiveIdx]
+  )
+
+  const handleNextClick = useCallback(() => {
+    if (isLastAnnouncement) {
+      onClose()
+      setLastSeenAnnouncement()
+    }
+
+    setCurrActiveIdx(Math.min(currActiveIdx + 1, announcements.length - 1))
+  }, [
+    announcements.length,
```

Comment:
Hey, `announcements.length` is redundant in this dependency array since `isLastAnnouncement`'s hook also conatains `announcements.length` as part of its dependency array

Reply:
this is true but it's auto added in as part of a rule (i think it's `useEffect exhaustive deps` or something to that effect) and it's because react isn't smart enough to figure out transitive deps (A -> B and B -> C therefore A -> C). 

you could try removing and saving, it should be auto added back in